### DISCUSSION TITLE
Is it possible to dynamically change harfbuzz_features?

### DISCUSSION THREAD

**Discussion by ianliu** [wezterm/wezterm#2515] [Q&A] [answered]
I would like to dynamically activate/deactivate harfbuzz features based on which file type I'm editing on VIM. Is this possible?

An example is to enable some ligatures when editing Haskell files (like `>>=`), but disable on my prompt, for instance.

Ideally this could be enabled/disable with shell escape sequences, to allow different regions of the terminal to behave differently!

**Comment by wez** [comment, marked as answer]
I'd suggest emitting a user var to reflect that preference.
https://wezfurlong.org/wezterm/config/lua/window-events/user-var-changed.html can be used to take an action when a user var is changed, and then you can use [window:set_config_overrides](https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html) to adjust the config accordingly.
